lane.py

```diff
@@ -1907,18 +1913,23 @@ def _groups_for_lanes(
         :yield: A sequence of (Work, WorkList) 2-tuples, with each
             WorkList representing the child WorkList in which the Work is
             found.
+
         """
         library = self.get_library(_db)
-        target_size = library.featured_lane_size
-
-        # We ask for a few extra works for each lane, to reduce the
-        # risk that we'll end up reusing a book in two different
-        # lanes.
-        ask_for_size = max(target_size+1, int(target_size * 1.10))
-        # TODO: we're reusing this pagination object, which means
-        # page_loaded will be called multiple times. Could this be a
-        # problem?
-        pagination = Pagination(size=ask_for_size)
+        if pagination is None:
+            # No pagination object was provided. Our target size is
+            # the featured lane size, but we'll ask for a few extra
+            # works for each lane, to reduce the risk that we end up
+            # reusing a book in two different lanes.
+            target_size = library.featured_lane_size
+
+            # We ask for a few extra works for each lane, to reduce the
+            # risk that we'll end up reusing a book in two different
+            # lanes.
+            ask_for_size = max(target_size+1, int(target_size * 1.10))
```

Comment:
Could you tell when it's possible?

Reply:
Sure, let's say you have a lane for "Romance" and a lane for "Fantasy". Since a book can be classified in multiple genres, it might happen that we randomly pick the same "romance fantasy" book for both lanes.